luigi/contrib/kubernetes.py

```diff
@@ -239,10 +249,19 @@ def signal_complete(self):
         pass
 
     def __get_pods(self):
-        pod_objs = Pod.objects(self.__kube_api, namespace=self.kubernetes_namespace) \
-            .filter(selector="job-name=" + self.uu_name) \
-            .response['items']
-        return [Pod(self.__kube_api, p) for p in pod_objs]
+        for _ in range(self.max_retrials_to_get_pods + 1):
+            pod_objs = Pod.objects(self.__kube_api, namespace=self.kubernetes_namespace) \
+                .filter(selector="job-name=" + self.uu_name) \
+                .response['items']
+            pods = [Pod(self.__kube_api, p) for p in pod_objs]
+
+            if pods:
+                break
+
+            # If pod was not returned, sleep to wait for pod to be created.
+            time.sleep(self.__POLL_TIME)
+
+        return pods
```

Comment:
am i correct if the pod isn't available this will return `None`? (just want to be sure i understand)

Reply:
Error logs:

```
Traceback (most recent call last):
  File "/Users/kumezawa/.pyenv/versions/anaconda3-4.3.0/lib/python3.6/site-packages/luigi/worker.py", line 200, in run
    new_deps = self._run_get_new_deps()
  File "/Users/kumezawa/.pyenv/versions/anaconda3-4.3.0/lib/python3.6/site-packages/luigi/worker.py", line 140, in _run_get_new_deps
    task_gen = self.task.run()
  File "/Users/kumezawa/.pyenv/versions/anaconda3-4.3.0/lib/python3.6/site-packages/luigi/contrib/kubernetes.py", line 400, in run
    self.__track_job()
  File "/Users/kumezawa/.pyenv/versions/anaconda3-4.3.0/lib/python3.6/site-packages/luigi/contrib/kubernetes.py", line 233, in __track_job
    while not self.__verify_job_has_started():
  File "/Users/kumezawa/.pyenv/versions/anaconda3-4.3.0/lib/python3.6/site-packages/luigi/contrib/kubernetes.py", line 302, in __verify_job_has_started
    pods = self.__get_pods()
  File "/Users/kumezawa/.pyenv/versions/anaconda3-4.3.0/lib/python3.6/site-packages/luigi/contrib/kubernetes.py", line 274, in __get_pods
    return pods
UnboundLocalError: local variable 'pods' referenced before assignment
```